src/main/kotlin/no/fdk/fdk_informationmodel_harvester/harvester/HarvesterActivity.kt

```diff
@@ -0,0 +1,60 @@
+package no.fdk.fdk_informationmodel_harvester.harvester
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.launch
+import no.fdk.fdk_informationmodel_harvester.adapter.HarvestAdminAdapter
+import no.fdk.fdk_informationmodel_harvester.rabbit.RabbitMQPublisher
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.util.*
+import javax.annotation.PostConstruct
+
+private val LOGGER = LoggerFactory.getLogger(HarvesterActivity::class.java)
+private const val HARVEST_ALL_ID = "all"
+
+@Service
+class HarvesterActivity(
+    private val harvestAdminAdapter: HarvestAdminAdapter,
+    private val harvester: InformationModelHarvester,
+    private val publisher: RabbitMQPublisher
+): CoroutineScope by CoroutineScope(Dispatchers.Default) {
+
+    @PostConstruct
```

Comment:
Kan du sjekke om vi kan bruke application listener i stedet (https://reflectoring.io/spring-boot-execute-on-startup/#applicationlistener)?

Jeg tror du kan da slippe Riot.init() i main fila

Reply:
Prøvde med application listener første gang jeg satte dette opp med datatjeneste-høsteren, fikk det ikke til å fungere slik jeg ville og PostConstruct fungerte helt greit. Så jeg ser ikke noe behov for å endre det.

RIOT.init() vil ha lite å si med dette. Vi kan kanskje fjerne den fra main() uansett, men jeg ser ikke verdien i å fjerne den når det potensielt innfører bugs og koster så lite å la stå...